buildtest/buildsystem/base.py

```diff
@@ -545,7 +516,6 @@ def _write_test(self, lines=None):
         # '$HOME/.buildtest/testdir/<name>/<name>_<timestamp>.sh'
         # This will put output (latest run) in same directory - do we want this?
         testpath = os.path.expandvars(self.metadata["testpath"])
```

Comment:
Should we save this with the builder so it doesn't need to be run (with expandvars) more than once? It should be done at the beginning of run/dry_run or something shared between them before that.

Reply:
hmm, that does make sense shall i overwrite the variable ``self.metadata["testpath"]``

Reply:
That would be easy enough to do - the reason to have the expansion is to honor some custom environment variable definition, but once the build has started the user can consider that "frozen" so I think updating it (and using that full path for the rest of tests) makes most sense.

Reply:
okay just to confirm with latest change it should be working. A test for variable expansion would be something like 

```
(buildtest-framework) ssi29@ag-mxg-hulk090> buildtest build -c examples/script/slurm.yml --testdir $PWD/dummy
Config Name                    SubTest                        Status                         Config Path
________________________________________________________________________________________________________________________
slurm                          slurm_down_nodes_reason        PASSED                         /mxg-hpc/users/ssi29/buildtest-framework/examples/script/slurm.yml
slurm                          slurm_not_responding_nodes     PASSED                         /mxg-hpc/users/ssi29/buildtest-framework/examples/script/slurm.yml
==============================================================
                         Test summary
Executed 2 tests
Passed Tests: 0 Percentage: 0.0%
Failed Tests: 2 Percentage: 100.0%
(buildtest-framework) ssi29@ag-mxg-hulk090> tree dummy/
dummy/
├── run
│   ├── slurm_down_nodes_reason_04-08-2020-21-25.err
│   ├── slurm_down_nodes_reason_04-08-2020-21-25.out
│   ├── slurm_not_responding_nodes_04-08-2020-21-25.err
│   └── slurm_not_responding_nodes_04-08-2020-21-25.out
├── slurm_down_nodes_reason.sh
└── slurm_not_responding_nodes.sh

1 directory, 6 files

```

Reply:
Why does it say there are 2 percentage failed tests (then 100%) but report them above as PASSED?

Reply:
For the first part I understand it’s a count followed by “Percentage 100%” so it’s just a confusing read for the user. Can we do:

2/2: 100% 

Or something simple like that!

Reply:
Let me look into this, good catch i didn't pay attention hold on

Reply:
okay this is fixed now i had typo again for some reason i was reporting test as PASSED

```
(buildtest-framework) ssi29@ag-mxg-hulk090> buildtest build -c examples/script/slurm.yml
Config Name                    SubTest                        Status                         Config Path
________________________________________________________________________________________________________________________
slurm                          slurm_down_nodes_reason        FAILED                         /mxg-hpc/users/ssi29/buildtest-framework/examples/script/slurm.yml
slurm                          slurm_not_responding_nodes     FAILED                         /mxg-hpc/users/ssi29/buildtest-framework/examples/script/slurm.yml
==============================================================
                         Test summary
Executed 2 tests
Passed Tests: 0/2 Percentage: 0.0%
Failed Tests: 2/2 Percentage: 100.0%

(buildtest-framework) ssi29@ag-mxg-hulk090> buildtest build -c examples/script/python-circle.yml
Config Name                    SubTest                        Status                         Config Path
________________________________________________________________________________________________________________________
python-circle                  circle_area                    PASSED                         /mxg-hpc/users/ssi29/buildtest-framework/examples/script/python-circle.yml
==============================================================
                         Test summary
Executed 1 tests
Passed Tests: 1/1 Percentage: 100.0%
Failed Tests: 0/1 Percentage: 0.0%

```

Reply:
Looks good!